### PR TITLE
[SPIKE] [CFP-410] FormBuilder migration - Travel expenses

### PR DESCRIPTION
**Pull request by NenehB**
#### What

Continue the migration away from the deprecated design system to the current GOVUK Frontend.

#### Ticket
[CFP-77-FormBuilder migration - Travel expenses (https://dsdmoj.atlassian.net/browse/CFP-410)

#### Why

- Update the service to use GOV.UK styles, components and patterns
- To improve and resolve the service's accessibility issues
- To reduce code complexity
- To improve maintainability

--------

#### TODO (wip)

- [x]  Update translation
- [x]  Update spec
- [x]  Update aob (link location for lgfs does not work)

### CODE REVIEW

File: app/views/external_users/claims/expenses/_expense_fields.html.haml [hunk 1]
```diff
@@ -1,134 +1,125 @@
...
+          -#   :description,
+          -#   form_group: { classes: 'fx-travel-mileage fx-travel-mileage-bike', style: 'display: none' },
+          -#   legend: { text: t('.cost_html') }
+
+          -# = f.govuk_collection_radio_buttons :mileage_rate_id,
+          -#   Expense::CAR_MILEAGE_RATES.values,
+          -#   :id,
+          -#   :description,
+          -#   form_group: { classes: 'fx-travel-mileage fx-travel-mileage-car', style: 'display: none' },
+          -#   legend: { text: t('.cost_html') }
```
Comment (naseberry) [inline app/views/external_users/claims/expenses/_expense_fields.html.haml]: Is this redundant?

File: config/locales/en.yml [hunk 2]
```diff
@@ -260,8 +260,6 @@ en:
               accepted: You must accept the terms and conditions
         expense:
```
Comment (naseberry) [inline config/locales/en.yml]: Nit: should this section be moved to `config/locales/en/models/claim.yml`?

File: features/claims/advocate/scheme_eleven/advocate_fixed_fee_claim_submit.feature [hunk 3]
```diff
@@ -49,4 +49,5 @@ Feature: Advocate tries to submit a claim for a Fixed fee (Appeal against convic
     Then I click "Continue" in the claim form
     And I should be in the 'Fixed fees' form page
     And I should see the error 'Total value claimed must be greater than £0.00'
+    And pause for 30 seconds
```
Comment (naseberry) [inline features/claims/advocate/scheme_eleven/advocate_fixed_fee_claim_submit.feature]: Nit: can we remove this step?

File: features/claims/advocate/scheme_nine/advocate_claim_draft_edit_submit.feature [hunk 4]
```diff
@@ -57,6 +57,7 @@ Feature: Advocate partially fills out a draft claim for a trial, then later edit
     And I add an expense date for scheme 9
 
     Then I click "Continue" in the claim form
+    And pause for 30 seconds
```
Comment (naseberry) [inline features/claims/advocate/scheme_nine/advocate_claim_draft_edit_submit.feature]: Nit: can we remove this step?

File: app/views/external_users/claims/expenses/_expense_fields.html.haml [hunk 5]
```diff
@@ -1,134 +1,110 @@
...
+                form_group: { classes: 'fx-establishment-select has-select', style: 'display: none' },
+                label: { text: nil }
+
+            = f.govuk_text_field :location,
+                class: 'fx-location-model',
+                form_group: { classes: 'location_wrapper' },
+                label: { text: t('.location_html') },
+                width: 'one-half'
+
+          = f.govuk_number_field :hours,
```
Comment (naseberry) [inline app/views/external_users/claims/expenses/_expense_fields.html.haml:61]: Nit: consider adding a `min:` value to  all `govuk_number_field` to limit errors

File: config/locales/en/models/claim.yml [hunk 6]
```diff
@@ -132,3 +132,49 @@ en:
 
             offence:
               blank: Choose an offence
+        expense:
```
Comment (naseberry) [inline config/locales/en/models/claim.yml]: Did we discuss normalizing this file as part of our formbuilder work?

File: app/validators/base_sub_model_validator.rb [hunk 7]
```diff
@@ -71,7 +71,7 @@ def associated_error_attribute(association_name, record_num, error)
     #
     # NOTE: Once form migrations are complete, this conditional can be removed
     #
-    if %i[defendants representation_orders dates_attended fixed_fees fixed_fee].include? association_name
+    if %i[defendants representation_orders dates_attended fixed_fees fixed_fee expenses].include? association_name
```
Comment (naseberry) [inline app/validators/base_sub_model_validator.rb]: Nit / Standard: this array should ideally be in alphabetical order

File: app/views/external_users/claims/_error_summary.html.haml [hunk 8]
```diff
@@ -1,4 +1,4 @@
-- form_steps = %w[case_details defendants fixed_fees transfer_fee_details offence_details supporting_evidence]
+- form_steps = %w[case_details defendants fixed_fees transfer_fee_details offence_details supporting_evidence travel_expenses]
```
Comment (naseberry) [inline app/views/external_users/claims/_error_summary.html.haml]: Nit / Standard: this array should ideally be in alphabetical order

File: app/views/external_users/claims/expenses/_textarea_additional_info.html.haml [hunk 9]
```diff
@@ -1,8 +1,3 @@
...
-  = t('.additional_info_heading')
-
-.form-group{ class: ('field_with_errors form-group-error' if validation_error_message(@error_presenter, 'travel_expense_additional_information').present?) }
-  %label.form-hint{ for: 'travel_expense_additional_information' }
-    = t(".#{@claim.agfs? ? 'agfs' : 'lgfs'}_additional_info_label")
-    = validation_error_message(@error_presenter, 'travel_expense_additional_information')
-  = f.text_area :travel_expense_additional_information, class: 'form-control form-control-full', id: 'travel_expense_additional_information', rows: 5
+= f.govuk_text_area :travel_expense_additional_information,
+  label: { text: t('.additional_info_heading'), size: 'm' },
+  hint: { text: t(".#{@claim.agfs? ? 'agfs' : 'lgfs'}_additional_info_label") }
```
Comment (naseberry) [inline app/views/external_users/claims/expenses/_textarea_additional_info.html.haml]: Nit / Standard: options should be in alphabetical order

File: app/views/external_users/claims/expenses/_expense_fields.html.haml [hunk 10]
```diff
@@ -1,134 +1,114 @@
...
+            reasons_collection.map { |et| [et.reason, et.id, { data: et.data_attributes }] },
+            :second,
+            :first,
+            form_group: { classes: 'fx-travel-reason', style: 'display: none' },
+            label: { text: t('.travel_reason_html') },
+            options: { include_blank: t('.select_option') }
+
+          - present(f.object) do |expense|
+            = f.govuk_text_field :reason_text,
+              form_group: { classes: 'fx-travel-reason-other', style: f.object.reason_id.eql?(5) ? 'display: block' : 'display: none' },
```
Comment (naseberry) [inline app/views/external_users/claims/expenses/_expense_fields.html.haml:42]: `f.object.reason_id.eql?(5) ? 'display: block' : 'display: none'` seems to be redundant code.
This ternary would only work on the server side, but this entire step uses client side script to work.

Note: I've created ticket (https://dsdmoj.atlassian.net/browse/CFP-614) to action the  progressive enhancement strategy.

**Review by naseberry — COMMENTED** [review]
Aside from the small nits I've identified, code changes look good.
I've sanity checked changes against a hosted environment, 👍 

I would recommend you get at least 2 approval on this before merging.

**Review by imtiazAhmd — APPROVED** [review]
LGTM 👍🏼

**Review by naseberry — APPROVED** [review]

**Pull request merged**